### DISCUSSION TITLE
OTA firmware for TRV HY368 (_TZE200_cpmgn2cf)?

### DISCUSSION THREAD

**Discussion by ra-kal** [Koenkk/zigbee2mqtt#15104] [Devices]
Hi,

I'm fighting a little bit with HY368 aka TS0601 thermostat, model  _TZE200_cpmgn2cf.
It works, but it often reports the values as unavailable in HA after some time (a day or so?). For example a part of the its state:

```
    "local_temperature_calibration": 0,
    "max_temperature": null,
    "min_temperature": null,
    "position": 75,
    "preset": "manual",
```

Therefore I thought I would check if the valve had the latest FW. but the FW version info is missing.
I tried to update it with OTA but it just ends with the following error: 

`Failed to check if update available for 'Thermostat A' (No image available for imageType '5634').`

Any help would be appreciated.

**Comment by ra-kal** [comment]
I found there exists firmare update for the TRV, but not for my model, which has different Manufacturer code.
https://github.com/Koenkk/zigbee-OTA/blob/master/index.json

```
        "fileVersion": 87,
        "fileSize": 234174,
        "manufacturerCode": 4098,
        "manufacturerName": [
            "_TZE200_ckud7u2l",
            "_TZE200_cwnjrr72",
            "_TZE200_ywdxldoj"
        ],
        "imageType": 5634,
        "sha512": "1113486a67403d922fc4a462b7477e0a0986fa8862a89b74ab8c5eac7b4ecb31c30f1da9ae1bbc1d6d50ae1fed66d38c37a1834ab71fd384c0555002783aba13",
        "url": "https://github.com/Koenkk/zigbee-OTA/raw/master/images/Tuya/1615190575-si32_zg_uart_connect_sleep_ZS5_ty_OTA_1.1.7.bin",
        "path": "images/Tuya/1615190575-si32_zg_uart_connect_sleep_ZS5_ty_OTA_1.1.7.bin"
```

**Reply by nizaga** [reply]
I am facing the same issue. I have 4 TRV, 3 of them have the manufacturer code _TZE200_cpmgn2cf and 1 has the manufacturer code _TZE200_cwnjrr72. I am able to upate one of them but not the others.

Is there a way to force an update?

**Reply by ra-kal** [reply]
I found out one can define local OTA index:
https://www.zigbee2mqtt.io/guide/usage/ota_updates.html#local-ota-index-and-firmware-files

So theoretically we should be able to copy the index.json file from the Zigbee-OTA repository to local disk, add the manufacturerName and tell Z2M to use this index.
But it can be risky, because we do not know, if out model is compatible with the FW. I assume yes, because probably only the new manufacturers were added to Z2M to support these models, but not into OTA list.

Actually I would like to see first, what version my TRV has. Do you see the FW version on the one you could update?

Btw, my problem with leaving devices seems to be more network related. But I'm still investigating it.

**Reply by nizaga** [reply]
I can't see the firmware version on the OTA section. I can't see the firmware date neither.

In the logs, i see the following when requesting for an update:  https://github.com/Koenkk/zigbee-OTA/raw/master/images/Tuya/1615190575-si32_zg_uart_connect_sleep_ZS5_ty_OTA_1.1.7.bin

**Reply by nizaga** [reply]
btw, my TRV are working fine (so far no issue found). However, i tried to update the FW and this i why i got the error

**Reply by NorthernMunkey** [reply]
did you update them in the end?

**Reply by ra-kal** [reply]
Not updated. My problems do not seem to be related with the sleeping issues that fixes the latest FW. The initial problems was resolved by better positioning the coordinator.
Now I have 25 zigbee devices and 4 thermostat valves. Only from one thermostat I do not get correct data time to time. It connects throuth a powered device, but even it is in the same room, it reports pretty low LQI, usually below 100. I tried multiple time to rejoin it, but nothing worked so far.

**Reply by yellowsta726** [reply]
I was able to flash firmware 1615190575-si32_zg_uart_connect_sleep_ZS5_ty_OTA_1.1.7.bin on TS0601 _TZE200_cpmgn2cf

Zigbee2MQTT Log level = Info  (the update works correctly if the info level is set)

my ota.json

```
[
    {
        "fileVersion": 87,
        "fileSize": 234174,
        "manufacturerCode": 4098,
        "manufacturerName": [
            "_TZE200_ckud7u2l",
            "_TZE200_cwnjrr72",
            "_TZE200_ywdxldoj",
            "_TZE200_cpmgn2cf"
        ],
        "imageType": 5634,
        "sha512": "1113486a67403d922fc4a462b7477e0a0986fa8862a89b74ab8c5eac7b4ecb31c30f1da9ae1bbc1d6d50ae1fed66d38c37a1834ab71fd384c0555002783aba13",
        "url": "1615190575-si32_zg_uart_connect_sleep_ZS5_ty_OTA_1.1.7.bin",
        "force": true
    }
]
```

ota.json and 1615190575-si32_zg_uart_connect_sleep_ZS5_ty_OTA_1.1.7.bin are in the same directory as file configuration.yaml

**Reply by yellowsta726** [reply]
https://github.com/Koenkk/zigbee-OTA/commit/f42834fb6bcce1a8d95b42e5902f7471a82cd171

**Comment by ra-kal** [comment]
Where can I find changelog for the FW versions?

```
    "update": {
        "installed_version": 85,
        "latest_version": 87,
        "state": "available"
    },
```

**Reply by freeDom-** [reply]
I am also interested in this, but I have the bad feelings that we won't find them at all. Currently I am trying to decide whether to install the firmware update or not.
